homeassistant/components/gardena_bluetooth/sensor.py

```diff
@@ -0,0 +1,125 @@
+
+from dataclasses import dataclass, field
+from datetime import datetime, timedelta, timezone
+
+from gardena_bluetooth.const import Battery, Valve
+from gardena_bluetooth.parse import Characteristic
+
+from homeassistant.components.sensor import (
+    SensorDeviceClass,
+    SensorEntity,
+    SensorEntityDescription,
+    SensorStateClass,
+)
+from homeassistant.config_entries import ConfigEntry
+from homeassistant.const import PERCENTAGE, EntityCategory
+from homeassistant.core import HomeAssistant
+from homeassistant.helpers.entity_platform import AddEntitiesCallback
+import homeassistant.util.dt as dt_util
+
+from .const import DOMAIN
+from .coordinator import Coordinator, GardenaBluetoothEntity
+
+
+@dataclass
+class GardenaBluetoothSensorEntityDescription(SensorEntityDescription):
+    """Description of entity."""
+
+    char: Characteristic = field(default_factory=lambda: Characteristic(""))
+
+
+DESCRIPTIONS = (
+    GardenaBluetoothSensorEntityDescription(
+        key=Valve.activation_reason.uuid,
+        translation_key="activation_reason",
+        state_class=SensorStateClass.MEASUREMENT,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        entity_registry_enabled_default=False,
+        char=Valve.activation_reason,
+    ),
+    GardenaBluetoothSensorEntityDescription(
+        key=Battery.battery_level.uuid,
+        state_class=SensorStateClass.MEASUREMENT,
+        device_class=SensorDeviceClass.BATTERY,
+        entity_category=EntityCategory.DIAGNOSTIC,
+        native_unit_of_measurement=PERCENTAGE,
+        char=Battery.battery_level,
+    ),
+)
+
+
+async def async_setup_entry(
+    hass: HomeAssistant, entry: ConfigEntry, async_add_entities: AddEntitiesCallback
+) -> None:
+    """Set up Gardena Bluetooth sensor based on a config entry."""
+    coordinator: Coordinator = hass.data[DOMAIN][entry.entry_id]
+    entities: list[GardenaBluetoothEntity] = [
+        GardenaBluetoothSensor(coordinator, description)
+        for description in DESCRIPTIONS
+        if description.key in coordinator.characteristics
+    ]
+    entities.append(GardenaBluetoothRemainSensor(coordinator))
+    async_add_entities(entities)
+
+
+class GardenaBluetoothSensor(GardenaBluetoothEntity, SensorEntity):
+    """Representation of a sensor."""
+
+    entity_description: GardenaBluetoothSensorEntityDescription
+
+    def __init__(
+        self,
+        coordinator: Coordinator,
+        description: GardenaBluetoothSensorEntityDescription,
+    ) -> None:
+        """Initialize the sensor."""
+        super().__init__(coordinator, {description.key})
+        self._attr_native_value = None
+        self._attr_unique_id = f"{coordinator.address}-{description.key}"
+        self.entity_description = description
+
+    def _handle_coordinator_update(self) -> None:
+        value = self.coordinator.read_cached(self.entity_description.char)
+        if isinstance(value, datetime):
+            value = value.replace(
```

Comment:
It would be better if the library provided times natively in utc so we don't have to convert here

Reply:
Device stores time in local time without knowledge about timezone.

Reply:
Suppose i could pass in timezone for the client on start. I will ponder a solution for future lib updates. I do agree